### ISSUE TITLE
CORS block keycloak sign-in

### ISSUE THREAD

**Issue by bo0ts** [MaibornWolff/SecObserve#2331]
I've configured Keycloak according to the documentation. When I click on the `Enterprise Sign In` button I get the following error message in the browser console:

```
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://keycloak.mydomain.intern/realms/mycompany/.well-known/openid-configuration. (Reason: CORS request did not succeed). Status code: (null).
```

I have the following configuration related to CORS

```
CORS_ALLOWED_ORIGINS: https://secobserve.mydomain.intern
```

The Keycloak version is 26.0.6. I assume we can fix this as soon as the multiple CORS Origins change is available? Maybe an addition to the documentation would be helpful as well.

**Comment by bo0ts** [comment]
After some investigation I suspect that this is actually caused by the private CA that is used for the keycloak domain. I'm looking for a way to easily inject a new CA in the backend image.

**Comment by StefanFl** [comment]
Ho @bo0ts , can we close this ticket or is there still something to do?

**Comment by bo0ts** [comment]
@StefanFl I don't think so. I've updated to 1.23.0 and have configured `CORS_ALLOWED_ORIGINS: "https://secobserve.mydomain.intern,https://keycloak.mydomain.intern"` in the backend.

I still get the same CORS error when clicking `Enterprise Sign In`.

**Comment by StefanFl** [comment]
The backend is not involved in this step. It happens between the frontend and keycloak. Several sources say, to add a `+` in the Web origins field in Keycloak's client administration:
![grafik](https://github.com/user-attachments/assets/142a6479-66e0-4db8-969c-407319db3202)

**Comment by bo0ts** [comment]
I've tried several options (include `+` and `*`) for the Web origins and this hasn't been necessary for any other client. Frankly, I don't see why it should. The request doesn't go to a client endpoint, but `https://keycloak.mydomain.intern/realms/mycompany/.well-known/openid-configuration` which isn't client aware.

Any other ideas where to look?

**Comment by StefanFl** [comment]
I tested it just now with [docker-compose-dev-keycloak.yml](https://github.com/MaibornWolff/SecObserve/blob/dev/docker-compose-dev-keycloak.yml). With the `+` in Web origins the login works:

![grafik](https://github.com/user-attachments/assets/174d956e-696a-4577-be54-f32a2a3f4641)

When I remove the `+` in Web origins and make this field empty, the login doesn't work anymore and I get this message in the browser console: `Quellübergreifende (Cross-Origin) Anfrage blockiert: Die Gleiche-Quelle-Regel verbietet das Lesen der externen Ressource auf http://localhost:8080/realms/secobserve/protocol/openid-connect/token. (Grund: CORS-Kopfzeile 'Access-Control-Allow-Origin' fehlt). Statuscode: 200.`

I am not a KeyCloak expert so I don't now anything more unfortunately.

**Comment by StefanFl** [comment]
@bo0ts Did you find a solution?

I will close this issue for now, but if you have any updates, feel free to open it again.

**Comment by StefanFl** [comment]
@bo0ts Did you find a solution?

I will close this issue for now, but if you have any updates, feel free to open it again.

**Comment by bo0ts** [comment]
@StefanFl Thanks for your patience with this. I could finally track this issue to Firefox with the uBlock Plugin enabled...
